packages/cli/src/commands/BaseCommand.ts

```diff
@@ -333,7 +333,9 @@ export abstract class BaseCommand extends Command {
 			this.logger.info(`Received ${signal}. Shutting down...`);
 			this.shutdownService.shutdown();
 
-			await Promise.all([this.stopProcess(), this.shutdownService.waitForShutdown()]);
+			await this.shutdownService.waitForShutdown();
+
+			await this.stopProcess();
```

Comment:
This also changes the shutdown behaviour of the `start` command. It seems to have some duplication (some functions are called in both `ShutdownService` and `StartCommand.stopProcess()`. We should fix that also as a follow-up